Sources/QuadTreeNode.swift

```diff
@@ -0,0 +1,83 @@
+import MapKit
+
+// MARK: - Tree node
+
+final class QuadTreeNode {
+
+  private let capacity: Int
+  private let rect: MKMapRect
+  private var children: [QuadTreeNode] = []
+  private var annotations: [MKAnnotation] = []
+
+  var isLeaf: Bool {
+    return children.isEmpty
+  }
+
+  init(rect: MKMapRect, capacity: Int) {
+    self.rect = rect
+    self.capacity = capacity
+  }
+
+  @discardableResult
+  func add(annotation: MKAnnotation) -> Bool {
+    guard rect.contains(annotation: annotation) else {
+      return false
+    }
+
+    if isLeaf {
+      annotations.append(annotation)
+      // Subdivide if needed
+      if annotations.count == capacity {
```

Comment:
@vadymmarkov just a question. So each node will have 4 annotations, and then it will create `child` to store other annotations?